### PR TITLE
Fix missing certificate management configurations

### PR DESCRIPTION
**Pull request by rene-dekker**
The certificate management feature has not kept up with recent changes to the operator. A number of issues were identified and patched:
- Missing RBAC for service accounts for newer components that have a tls secret
- Missing security context
- Missing init containers for newer tls secrets

### CODE REVIEW

File: pkg/controller/compliance/compliance_controller.go [hunk 1]
```diff
@@ -475,7 +475,7 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 	}
 	certificateComponent := rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
 		Namespace:       render.ComplianceNamespace,
-		ServiceAccounts: []string{render.ComplianceServerServiceAccount},
+		ServiceAccounts: []string{render.ComplianceServerServiceAccount, render.ComplianceBenchmarkerServiceAccount, render.ComplianceSnapshotterServiceAccount, render.ComplianceControllerServiceAccount},
```
Comment (asincu) [inline pkg/controller/compliance/compliance_controller.go:478]: What about reporter service account ?

File: pkg/render/compliance.go [hunk 2]
```diff
@@ -1021,6 +1027,10 @@ func (c *complianceComponent) complianceSnapshotterDeployment() *appsv1.Deployme
 				MountPath: LinseedVolumeMountPath,
 			})
 	}
+	var initContainers []corev1.Container
```
Comment (asincu) [inline pkg/render/compliance.go:1030]: How does this work for compliance reporter ?

File: pkg/render/compliance_test.go [hunk 3]
```diff
@@ -533,9 +533,23 @@ var _ = Describe("compliance rendering tests", func() {
...
+
 			complianceTLS, err := certificateManager.GetOrCreateKeyPair(cli, render.ComplianceServerCertSecret, common.OperatorNamespace(), []string{""})
 			Expect(err).NotTo(HaveOccurred())
 			cfg.ServerKeyPair = complianceTLS
+
+			controllerKeyPair, err := certificateManager.GetOrCreateKeyPair(cli, render.ComplianceControllerSecret, common.OperatorNamespace(), []string{""})
+			Expect(err).NotTo(HaveOccurred())
+			cfg.ControllerKeyPair = controllerKeyPair
+
+			cenchmarkerKeyPair, err := certificateManager.GetOrCreateKeyPair(cli, render.ComplianceBenchmarkerSecret, common.OperatorNamespace(), []string{""})
```
Comment (asincu) [inline pkg/render/compliance_test.go:545]: Typo: benchmarkerKeyPair

File: pkg/render/intrusion_detection.go [hunk 4]
```diff
@@ -692,6 +692,10 @@ func (c *intrusionDetectionComponent) deploymentPodTemplate() *corev1.PodTemplat
 		}
 		container.Env = append(container.Env, envVars...)
 	}
+	var initContainers []corev1.Container
+	if c.cfg.IntrusionDetectionCertSecret != nil && c.cfg.IntrusionDetectionCertSecret.UseCertificateManagement() {
+		initContainers = append(initContainers, c.cfg.IntrusionDetectionCertSecret.InitContainer(IntrusionDetectionNamespace))
```
Comment (asincu) [inline pkg/render/intrusion_detection.go:697]: Do we need to enhance our tests to use check for init containers ?

File: pkg/render/logstorage.go [hunk 5]
```diff
@@ -753,7 +755,8 @@ func (es elasticsearchComponent) podTemplate() corev1.PodTemplateSpec {
 					"memory": resource.MustParse("50Mi"),
 				},
 			},
-			SecurityContext: securitycontext.NewRootContext(false),
+			// Without a root context, it is not able to ln and chown.
+			SecurityContext: securitycontext.NewRootContext(true),
```
Comment (asincu) [inline pkg/render/logstorage.go:759]: Should we enhance our tests to check for init containers ?

File: pkg/controller/compliance/compliance_controller.go (hunk 1 above)
Comment (rene-dekker) [reply pkg/controller/compliance/compliance_controller.go:478]: You are right, I missed that.

File: pkg/render/compliance.go (hunk 2 above)
Comment (rene-dekker) [reply pkg/render/compliance.go:1035]: Added, thanks

**Review by asincu — APPROVED** [review]

**Review by Brian-McM — APPROVED** [review]

**Pull request merged**